gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-heartbeat/src/main/java/io/gravitee/gateway/services/heartbeat/impl/standalone/StandaloneHeartbeatStrategy.java

```diff
@@ -0,0 +1,145 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StandaloneHeartbeatStrategy.class);
+
+    private ExecutorService executorService;
+    private final EventRepository eventRepository;
+    private final int delay;
+    private final TimeUnit unit;
+    private Event heartbeatEvent;
+
+    // We make this field protected to be able to test easily without relying on ExecutorService
+    @VisibleForTesting
+    protected Event eventToCreate;
+
+    public StandaloneHeartbeatStrategy(HeartbeatDependencies heartbeatDependencies) {
+        this.delay = heartbeatDependencies.getDelay();
+        this.unit = heartbeatDependencies.getUnit();
+        this.eventRepository = heartbeatDependencies.getEventRepository();
+    }
+
+    @Override
+    public void doStart(Event eventToCreate) throws Exception {
+        LOGGER.info("Start gateway heartbeat service");
+
+        this.eventToCreate = eventToCreate;
+
+        executorService = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "gio-heartbeat"));
+
+        LOGGER.info("Gateway heartbeat service scheduled with fixed delay {} {}", delay, unit);
+
+        ((ScheduledExecutorService) executorService).scheduleWithFixedDelay(this::sendHeartbeatEvent, 0, delay, unit);
+
+        LOGGER.info("Start gateway heartbeat service: DONE");
+    }
+
+    @Override
+    public void preStop() {
+        if (heartbeatEvent != null) {
+            sendStopHeartbeatEvent();
+        }
+        LOGGER.debug("Pre-stopping gateway heartbeat service");
+    }
+
+    @Override
+    public void doStop() {
+        if (!executorService.isShutdown()) {
+            LOGGER.info("Stop gateway heartbeat service");
+            executorService.shutdownNow();
+        } else {
+            LOGGER.info("Gateway heartbeat service already shut-downed");
+        }
+        LOGGER.info("Stop gateway heartbeat service: DONE");
+    }
+
+    @VisibleForTesting
+    void sendHeartbeatEvent() {
+        LOGGER.debug("Sending heartbeat event");
+
+        try {
+            if (heartbeatEvent == null) {
+                heartbeatEvent = eventRepository.createOrUpdateHeartbeat(eventToCreate);
+            } else {
+                // If heartbeatEvent is known, it means it has already been saved in database, so we can update it with a lite version
+                eventRepository.createOrUpdateHeartbeat(getLiteEvent(heartbeatEvent));
+            }
+        } catch (Exception ex) {
+            LOGGER.warn(
+                "An error occurred while trying to create the heartbeat event id[{}] type[{}]",
+                eventToCreate.getId(),
+                eventToCreate.getType(),
+                ex
+            );
+        }
+    }
+
+    /**
+     * Provides a lite version of the Heartbeat Event to only update date related fields
+     * @param heartbeatEvent the original heartbeat event
+     * @return the lite database
+     */
+    private Event getLiteEvent(Event heartbeatEvent) {
```

Comment:
The `get` naming makes me think it retreives an event.
But at all, this just build an event object.
Naming suggestion : `buildLiteEvent`